File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text1);
int count_words(string text1);
int count_sentences(string text1);

int main(void)
{
    string text = get_string("Text: ");
    int letters, words, sentences;
    float index, L, S;
    letters = count_letters(text);
    words = count_words(text);
    sentences = count_sentences(text);


    L = (float) letters / (float) words * 100;
    S = (float) sentences / (float) words * 100;
    index = 0.0588 * L - 0.296 * S - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %.0f\n", round(index));
    }
}

int count_letters(string text1)
{
    int letters = 0;




    for (int i = 0; i < strlen(text1); i++)
    {
        if (isalpha(text1[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text1)
{
    int words = 0;




    for (int i = 0; i < strlen(text1); i++)
    {
        if (text1[i] == ' ')
        {
            words ++;
        }
    }
    if (strlen(text1) > 0)
    {
        words++;
    }

    return words;
}

int count_sentences(string text1)
{
    int sentences = 0;




    for (int i = 0; i < strlen(text1); i++)
    {
        if (text1[i] == '.' || text1[i] == '!' || text1[i] == '?')
        {
            sentences ++;

        }
    }

    return sentences;

}

File: sentimental-readability/readability.py
# TODO
from cs50 import get_string

text = get_string("Text: ")

words = 0
sentences = 0
letters = 0
i = 0
length = len(text)

while i < length:
    if text[i].isalpha():
        letters = letters + 1
    if (i == 0 and text[i] != " ") or (i != length and text[i] == " " and text[i + 1] != " "):
        words = words + 1
    if text[i] == "." or text[i] == "?" or text[i] == "!":
        sentences = sentences + 1
    i = i + 1

L = (letters / words) * 100
S = (sentences / words) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")